documentation/metrics_api.markdown

```diff
@@ -0,0 +1,267 @@
+---
+layout: default
+title: "Puppet Server: Metrics API"
+canonical: "/puppetserver/latest/metrics_api.html"
+---
+
+# The Metrics API
+
+Puppet Enterprise includes two optional, enabled-by-default web APIs for
+[Java Management Extension (JMX)](https://docs.oracle.com/javase/tutorial/jmx/index.html)
+metrics, namely
+[managed beans (MBeans)](https://docs.oracle.com/javase/tutorial/jmx/mbeans/).
+
+## Metrics v2
```

Comment:
Hmm, so one thing I think we should talk about here is how to enable the Jolokia version of the Metrics endpoint - since it isn't enabled by default, from what I remember.  This [doc](https://github.com/puppetlabs/trapperkeeper-metrics/blob/7e6d2bf6a1c61f1c652405492c402ee1186ea860/documentation/configuration.md) in the trapperkeeper-metrics repo has a section about the `metrics-webservice.jolokia` configuration in it.  It doesn't appear that we updated the default [metrics.conf](https://github.com/puppetlabs/puppetserver/blob/bd05e6ef2ca0229c82526928843085bd50f0e58f/ezbake/config/conf.d/metrics.conf) file in Puppet Server with the skeleton of this configuration section, unfortunately.  I actually think it might be good to do that as part of this work - adding in a config where jolokia is disabled by default.  Could be done as a separate PR from this one if @gguillotte is working on his own revision to these docs and doesn't want us to merge this.